model/optimize/image.go

```diff
@@ -120,16 +122,18 @@ func (i *Image) Optimize(objects []core.PdfObject) (optimizedObjects []core.PdfO
 		if err != nil {
 			return nil, err
 		}
-		newStream := &core.PdfObjectStream{Stream: streamData}
-		newStream.PdfObjectReference = stream.PdfObjectReference
-		newStream.PdfObjectDictionary = core.MakeDict()
-		newStream.PdfObjectDictionary.Merge(stream.PdfObjectDictionary)
-		fn := core.PdfObjectName(encoder.GetFilterName())
-		newStream.PdfObjectDictionary.Set(core.PdfObjectName("Filter"), &fn)
-		ln := core.PdfObjectInteger(int64(len(streamData)))
-		newStream.PdfObjectDictionary.Set(core.PdfObjectName("Length"), &ln)
-		replaceTable[stream] = newStream
-		images[index].Stream = newStream
+		if int64(len(streamData)) < img.OrigSize {
```

Comment:
The new `OrigSize` field does not seem necessary as it can be obtained here. Also using `continue` here does seem suitable instead of adding an additional indentation level to that code block. Maybe we could also add a `Debug()` log message before `continue` in order to observe the cases in which this scenario occurs.

```
originalSize := len(stream.Stream)
if originalSize < len(streamData) {
    continue
}
```

Reply:
Good point, will fix.